pkg/cmd/drt-run/operations.go

```diff
@@ -108,54 +108,53 @@ func (r *opsRunner) pickOperation(
 		setIdx := rng.Intn(len(r.specs))
 		opSpecsSet := r.specs[setIdx]
 		opSpec := &opSpecsSet[rng.Intn(len(opSpecsSet))]
-		r.mu.Lock()
-		if r.mu.lockOutOperations {
-			r.mu.completed.Wait()
-			r.mu.Unlock()
-			continue
-		}
+		func() {
+			r.mu.Lock()
+			defer r.mu.Unlock()
+			if r.mu.lockOutOperations {
+				r.mu.completed.Wait()
+				return
+			}
 
-		lastRun := r.mu.lastRun[opSpec.Name]
-		eligibleForNextRun := lastRun.Add(r.config.Operations.Sets[setIdx].Cadence)
+			lastRun := r.mu.lastRun[opSpec.Name]
+			eligibleForNextRun := lastRun.Add(r.config.Operations.Sets[setIdx].Cadence)
 
-		if timeutil.Now().Compare(eligibleForNextRun) < 0 {
-			// Find another operation to run.
-			r.mu.completed.Wait()
-			r.mu.Unlock()
-			continue
-		}
-		// Ratchet lastRun forward.
-		r.mu.lastRun[opSpec.Name] = timeutil.Now()
-		r.mu.runningOperations[workerIdx] = opSpec.Name
-
-		// See what level of isolation this operation requires
-		// from other operations.
-		switch opSpec.CanRunConcurrently {
-		case registry.OperationCanRunConcurrently:
-			// Nothing to do.
-		case registry.OperationCannotRunConcurrently:
-			r.mu.lockOutOperations = true
-			fallthrough
-		case registry.OperationCannotRunConcurrentlyWithItself:
-			for otherOpsRunning := true; otherOpsRunning; {
-				otherOpsRunning = false
-				for i := range r.mu.runningOperations {
-					if i == workerIdx {
-						continue
+			if timeutil.Now().Compare(eligibleForNextRun) < 0 {
+				// Find another operation to run.
+				r.mu.completed.Wait()
+				return
+			}
+			// Ratchet lastRun forward.
+			r.mu.lastRun[opSpec.Name] = timeutil.Now()
+			r.mu.runningOperations[workerIdx] = opSpec.Name
+
+			// See what level of isolation this operation requires
+			// from other operations.
+			switch opSpec.CanRunConcurrently {
+			case registry.OperationCanRunConcurrently:
+				// Nothing to do.
+			case registry.OperationCannotRunConcurrently:
+				r.mu.lockOutOperations = true
+				fallthrough
+			case registry.OperationCannotRunConcurrentlyWithItself:
+				for otherOpsRunning := true; otherOpsRunning; {
+					otherOpsRunning = false
+					for i := range r.mu.runningOperations {
+						if i == workerIdx {
+							continue
+						}
+						if r.mu.runningOperations[i] != "" &&
+							(opSpec.CanRunConcurrently != registry.OperationCannotRunConcurrentlyWithItself || r.mu.runningOperations[i] == opSpec.Name) {
+							otherOpsRunning = true
+							break
+						}
 					}
-					if r.mu.runningOperations[i] != "" &&
-						(opSpec.CanRunConcurrently != registry.OperationCannotRunConcurrentlyWithItself || r.mu.runningOperations[i] == opSpec.Name) {
-						otherOpsRunning = true
-						break
+					if otherOpsRunning {
+						r.mu.completed.Wait()
 					}
 				}
-				if otherOpsRunning {
-					r.mu.completed.Wait()
-				}
 			}
-		}
-
-		r.mu.Unlock()
+		}()
 		return opSpec
```

Comment:
Some of the cases where were `continue`-ing in the loop earlier are no longer logically equivalent; we'll be returning from the closure and then returning the opSpec right here even if it's ineligible to be run. You might wanna add a return value to the function.

Reply:
I initially added a boolean and then realised that, this has the entire code, so, it was same. But as I can understand from what you are saying, if someone adds more code post this function, the continue will not work as expected. Will add a boolean return.